documentation/source/CONTRIBUTING.md

```diff
@@ -1,88 +1,101 @@
-
-    * Install pre-commit from pypi
-   ```
-    $ pip install pre-commit
-   ```    
-    * Set up pre-commit inside super-gradients repo which will create .git/hooks directory.
-   ```
-   $ pre-commit install
-   ```
-   ```
-   $ git commit -m "message"
-   ```
-   
-    * Each time you commit, git will run the pre-commit hooks (black and flake8 for now) on any python files that are getting committed and are part of the git index.  If black modifies/formats the file, or if flake8 finds any linting errors, the commit will not succeed. You will need to stage the file again if black changed the file, or fix the issues identified by flake8 and and stage it again.
-
-    * To run pre-commit on all files just run
-   ```
-   $ pre-commit run --all-files
-   ```
--->
-7. Create a pull request against <b>master</b> branch.
 
 
+## Code Style
 
-## Jupyter Notebooks Contribution
+We follow the **reStructuredText** docstring format (default of PyCharm), along with typing.
+```python
+def python_function(first_argument: int, second_argument: int) -> str:
+    """Do something with the two arguments.
 
-Pulling updates from remote might cause merge conflicts with jupyter notebooks. The tool [nbdime](https://nbdime.readthedocs.io/en/latest/) might solve this.
-* Installing nbdime
-```
-pip install ndime
-```
-* Run a diff between two notebooks
-```
-nbdiff notebook_1.ipynb notebook_2.ipynb
+    :param first_argument: First argument to the function
+    :param second_argument: Second argument to the function
+    :return: Description of the output
+    """
 ```
 
-## Code Style Guidelines
 
-We are working hard to make sure all the code in this repository is readable, maintainable and testable.
-We follow the Google docstring guidelines outlined on this [styleguide](https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings) page. For example:
-```python
-  def python_function(first_argument: int, second_argument: int) -> bool:
-      """
-      This function will do something with the two arguments.
 
-      Args:
-          first_argument (int): the first argument to the function
-          second_argument (int): the second argument to the function
 
-      Raises:
-          Exception: arguments should be ints not floats
+## Code Formatting
+
+We enforce [black](https://github.com/psf/black) code formatting in addition to existing flake8 checks. 
+
+To ensure everyone uses same code style, a project-wise [configuration file](https://github.com/Deci-AI/super-gradients/blob/master/pyproject.toml) has been added to SG repo. It ensures all formatting will be exactly the same regardless of OS, python version or the place where code formatting check is happening. 
+
```

Comment:
We also enforce flake8 lint.

Reply:
Added link in b3f061ac6b3420c2012ccb3af7c7cf8a86a0da82

Reply:
@Louis-Dupont can you please add the snippet to install git pre-commit hooks? That would help users to make pretty commits on their side.

Reply:
Do you mean this or something else ? https://github.com/Deci-AI/super-gradients/blob/fa86e97436625326034b7f762234ec520f7c7ed0/documentation/source/CONTRIBUTING.md?plain=1#L46-L49